packages/modules/providers/notification-sendgrid/src/services/sendgrid.ts

```diff
@@ -58,14 +58,28 @@ export class SendgridNotificationService extends AbstractNotificationProviderSer
 
     const from = notification.from?.trim() || this.config_.from
 
+    let mailContent
+
+    if ("providerContext" in notification) {
```

Comment:
```suggestion
    if ("providerContext" in notification && !!notification.providerContext) {
```

just to be sure if an empty object is provided maybe we want to handle it wdyt? because the `in` operator only ensure the key is present in the object, but it can still be undefined for example

Reply:
Good catch.